operators/src/pro/cache/cache_tiles.rs

```diff
@@ -296,14 +270,35 @@ impl<T> CacheTileStream<T> {
     }
 }
 
-impl<T: Pixel> Stream for CacheTileStream<T> {
-    type Item = Result<RasterTile2D<T>>;
+impl<T: Pixel> Stream for CacheTileStream<T>
+where
+    RasterTile2D<T>: CacheElement<StoredCacheElement = CompressedRasterTile2D<T>>,
+{
+    type Item = Result<RasterTile2D<T>, CacheError>;
 
     fn poll_next(
         mut self: Pin<&mut Self>,
-        _cx: &mut std::task::Context<'_>,
+        cx: &mut std::task::Context<'_>,
     ) -> std::task::Poll<Option<Self::Item>> {
-        let CacheTileStreamProjection { data, query, idx } = self.as_mut().project();
+        let CacheTileStreamProjection {
+            data,
+            query,
+            idx,
+            mut state,
+        } = self.as_mut().project();
+
+        // if there is a tile that is currently being decoded, try to return it
+        if let Some(state_future) = state.as_mut().as_pin_mut() {
```

Comment:
can you not put this logic into a `and_then` function on old CacheTileStream? Then you would not have to handle the future yourself.

Reply:
i can't do that. Introducing a lifetime of the &'a CompressedRasterTile2D<T>  that is needed to return a reference as stream item is to complex.